core/execution/future/market.go

```diff
@@ -2911,6 +3151,9 @@ func (m *Market) AmendOrderWithIDGenerator(
 
 	conf, updatedOrders, err := m.amendOrder(ctx, orderAmendment, party)
 	if err != nil {
+		if m.getMarginMode(party) == types.MarginModeIsolatedMargin && err == common.ErrMarginCheckFailed {
```

Comment:
So, is when trying to amend 1 order, we now fail the isolated margin check, we are cancelling ALL the orders of the party? that doesn't seounds right to me is it?